src/Compilers/CSharp/Portable/Syntax/Syntax.xml

```diff
@@ -1779,29 +1779,29 @@
   <Node Name="RecursivePatternSyntax" Base="PatternSyntax">
     <Kind Name="RecursivePattern" />
     <Field Name="Type" Type="TypeSyntax" Optional="true" />
-    <Field Name="DeconstructionSubpattern" Type="DeconstructionSubpatternSyntax" Optional="true" />
-    <Field Name="PropertySubpattern" Type="PropertySubpatternSyntax" Optional="true" />
+    <Field Name="DeconstructionPatternClause" Type="DeconstructionPatternClauseSyntax" Optional="true" />
+    <Field Name="PropertyPatternClause" Type="PropertyPatternClauseSyntax" Optional="true" />
     <Field Name="Designation" Type="VariableDesignationSyntax" Optional="true">
       <Kind Name="SingleVariableDesignation"/>
       <Kind Name="DiscardDesignation"/>
     </Field>
   </Node>
-  <Node Name="DeconstructionSubpatternSyntax" Base="CSharpSyntaxNode">
-    <Kind Name="DeconstructionSubpattern"/>
+  <Node Name="DeconstructionPatternClauseSyntax" Base="CSharpSyntaxNode">
+    <Kind Name="DeconstructionPatternClause"/>
     <Field Name="OpenParenToken" Type="SyntaxToken">
       <Kind Name="OpenParenToken"/>
     </Field>
-    <Field Name="SubPatterns" Type="SeparatedSyntaxList&lt;SubpatternElementSyntax&gt;"/>
+    <Field Name="Subpatterns" Type="SeparatedSyntaxList&lt;SubpatternElementSyntax&gt;"/>
```

Comment:
I found the name of this property a little confusing - this node itself is a subpattern and we're getting the individual "subpattern elements". Maybe this could just be called "Elements"?

Reply:
I agree. SubpatternElement implies two levels of parents. Looking at other syntax, `Elements:List<PatternElementSyntax>`could work out just fine but it wouldn't be as self-explanatory.

Reply:
actually, if "Subpattern" is renamed to "PatternClause", should "SubpatternElement" be "PatternClauseElement"?